TruePath.Tests/PathExtensionsTests.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TruePath.Tests;
+public class PathExtensionsTests
+{
+    [Theory]
+    [InlineData("foo/bar.txt", ".txt")]
+    [InlineData("/foo/bar.txt", ".txt")]
+    [InlineData("foo/bar.", "")]
+    [InlineData("foo/bar", null)]
+    public void GetExtensionWithDotTests(string path, string? expected)
+    {
+        IPath l = new LocalPath(path);
+        var pathResult = l.GetExtensionWithDot();
+        var value = l.FileName;
+        Assert.Equal(expected,  l.GetExtensionWithDot());
```

Comment:
I don't understand where `value` is used.
I also don't understand why the `GetExtensionWithDot` method should be called twice.

```suggestion
        IPath sut = new LocalPath(path);
        Assert.Equal(expected,  sut.GetExtensionWithDot());
```